command/build.go

```diff
@@ -270,16 +269,29 @@ func (c *BuildCommand) RunContext(buildCtx context.Context, cla *BuildArgs) int
 				runArtifacts,
 				err)
 			if hcperr != nil {
-				writeDiags(c.Ui, nil, hcl.Diagnostics{
-					&hcl.Diagnostic{
-						Summary: fmt.Sprintf(
-							"publishing build metadata to HCP Packer for %q failed",
-							name),
-						Severity: hcl.DiagError,
-						Detail:   hcperr.Error(),
-					},
-				})
-				hasPossibleIncompatibleHCPIntegration = true
+				if _, ok := hcperr.(*registry.NotAHCPArtifactError); ok {
```

Comment:
I was confused a bit because I thought hcperr meant an error specific to decoding but I now see it also may include an error if the build did not complete (aka no artifact generated).